plutus-chain-index/app/Marconi/Logging.hs

```diff
@@ -0,0 +1,122 @@
+    -- the most efficient structure but UTCTime is also not the most
+    -- efficient type.
+    numBlocks :: Int
+    numBlocks = 10
+
+    minSecondsBetweenMsg :: NominalDiffTime
+    minSecondsBetweenMsg = 10
+
+    update :: IORef SyncStats -> ChainSyncEvent (BlockInMode CardanoMode) -> IO ()
+    update statsRef (RollForward bim ct) = do
+      let cp = case bim of (BlockInMode (Block (BlockHeader slotNo hash _blockNo) _txs) _eim) -> ChainPoint slotNo hash
+      now <- getCurrentTime
+      modifyIORef' statsRef $ \stats ->
+        stats
+          { syncStatsAppliedBlocks = syncStatsAppliedBlocks stats + 1,
+            syncStatsLastAppliedBlocks = take numBlocks (now : syncStatsLastAppliedBlocks stats)
+          }
+      printMessage statsRef cp ct
+    update statsRef (RollBackward cp ct) = do
+      now <- getCurrentTime
+      modifyIORef' statsRef $ \stats -> stats {syncStatsLastRollback = Just now}
+      printMessage statsRef cp ct
+
+    printMessage statsRef cp ct = do
+      SyncStats {syncStatsAppliedBlocks, syncStatsLastAppliedBlocks, syncStatsLastRollback, syncStatsLastMessage} <- readIORef statsRef
+
+      let syncStatus = case (cp, ct) of
+            (_, ChainTipAtGenesis) ->
+              NotSynchronising
+            (ChainPointAtGenesis, _) ->
+              Synchronising 0
+            (ChainPoint (SlotNo chainPointSlot) _, ChainTip (SlotNo chainTipSlot) _header _blockNo)
+              -- TODO: MAGIC number here. Is there a better number?
+              -- 100 represents the number of slots before the
+              -- node where we consider the chain-index to be synced.
+              | chainTipSlot - chainPointSlot < 100 -> Synchronised
+            (ChainPoint (SlotNo chainPointSlot) _, ChainTip (SlotNo chainTipSlot) _header _blockNo) ->
+              let pct = (100 :: Double) * fromIntegral chainPointSlot / fromIntegral chainTipSlot
+               in Synchronising pct
+
+      let rate = case syncStatsLastAppliedBlocks of
+            _ : _ : _ ->
+              let timespan = diffUTCTime (List.head syncStatsLastAppliedBlocks) (List.last syncStatsLastAppliedBlocks)
+               in Just (fromIntegral numBlocks / realToFrac timespan :: Double)
+            _ ->
+              Nothing
+
+      let currentTipMsg = case syncStatus of
+            NotSynchronising -> ""
+            _                -> " Current point is " <> show cp <> "."
+
+      let rateMsg = case rate of
+            Nothing -> ""
+            Just r  -> " Current rate: " <> printf "%.2g" r <> " blocks/s."
+
+      let syncMsg = case syncStatus of
+            NotSynchronising -> " Not synchronising."
+            Synchronising p  -> printf " Synchornising (%0.2f%%)." p
+            Synchronised     -> " Synchronised."
+
+      let rollbackMsg = case syncStatsLastRollback of
+            Nothing -> ""
+            Just t  -> " Last rollback was on " <> show t <> "."
```

Comment:
Format the time. Currently it shows: `2022-05-31 17:50:03.393974842`, but I think it should be something like `2022-05-31T17:50:03Z` (or similar)